src/main/java/htsjdk/samtools/BAMFileReader2.java

```diff
@@ -0,0 +1,1205 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.seekablestream.SeekableStream;
+import htsjdk.samtools.util.*;
+import htsjdk.samtools.util.zip.InflaterFactory;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/** Class for reading and querying BAM files. */
+public class BAMFileReader2 extends SamReader.ReaderImplementation {
```

Comment:
This looks like a huge class to be used only one method (`BAMFileReader2.findVirtualOffsetOfFirstRecord(seekableStream)` in  in `BAMSBIIndexer.findVirtualOffsetOfFirstRecord`

Reply:
It is, however it's a temporary workaround until the [upstream PR](https://github.com/samtools/htsjdk/pull/1138) is available.